unstract/workflow-execution/src/unstract/workflow_execution/execution_file_handler.py

```diff
@@ -111,8 +128,14 @@ def add_metadata_to_volume(self, input_file_path: str, source_hash: str) -> None
             MetaDataKey.WORKFLOW_ID: str(self.workflow_id),
             MetaDataKey.EXECUTION_ID: str(self.execution_id),
         }
-        with fsspec.open(f"file://{destination_path}", "w") as local_file:
-            json.dump(content, local_file)
+        if check_feature_flag_status(FeatureFlag.REMOTE_FILE_STORAGE):
+            file_system = FileSystem(FileStorageType.WORKFLOW_EXECUTION)
+            file_storage = file_system.get_file_storage()
+            file_storage.write(path=metadata_path, mode="w", data=json.dumps(content))
```

Comment:
There is a FileStorage API json_dump() which may be doing this json dumping. Pls check if that can be used.

Reply:
Will you be taking care of this now or later?

Reply:
will be on later